src/main/java/com/friendly_machines/intellij/plugins/ideanative2debugger/impl/DebugProcess.java

```diff
@@ -257,102 +257,95 @@ private SuspendContext generateSuspendContext(List<Object> threads, String curre
         return new SuspendContext(this, stacks.toArray(new ExecutionStack[0]), activeStackId);
     }
 
-    public List<Map<String, Object>> getVariables(String threadId, String frameId) throws GdbMiOperationException, ClassCastException {
-        ArrayList<Map<String, Object>> result = new ArrayList<>();
+    public List<Map<String, ?>> getVariables(String threadId, String frameId) throws GdbMiOperationException, ClassCastException, IOException, InterruptedException {
         // TODO: --simple-values and find stuff yourself.
-        var q = gdbCall("-stack-list-variables", new String[]{"--thread", threadId, "--frame", frameId, "--all-values"}, new String[]{});
-        if (q.containsKey("variables")) {
-            try {
-                List<?> variables = (List<?>) q.get("variables");
-                for (Object variable1 : variables) {
-                    @SuppressWarnings("unchecked")
-                    var variable = (Map<String, Object>) variable1;
-                    result.add(variable);
-                }
-            } catch (ClassCastException e) {
-                e.printStackTrace();
+        var q = gdbCall("-stack-list-variables", List.of("--thread", threadId, "--frame", frameId, "--all-values"), Collections.emptyList());
+        try {
+            var variables = (Collection<Map<String, ?>>) q.get("variables");
+            if (variables == null) {
+                return Collections.emptyList();
             }
+            // pucgenie: Why even copy?
+            return new ArrayList<Map<String, ?>>(variables);
+        } catch (ClassCastException e) {
```

Comment:
ClassCastException is both caught and in our function signature. Maybe remove `catch` block ?

Reply:
I'd remove the throw and the catch because a ClassCastException shouldn't happen there, and if it happens, it will be just as effective as the current `throw RuntimeException`.